src/auth/authclient.ts

```diff
@@ -523,6 +511,106 @@ export abstract class AuthClient
       },
     };
   }
+
+  /**
+   * Refreshes trust boundary data for an authenticated client.
+   * Handles caching checks and potential fallbacks.
+   * @param tokens The refreshed credentials containing access token to call the trust boundary endpoint.
+   * @returns A Promise resolving to TrustBoundaryData or empty-string for no-op trust boundaries.
+   * @throws {Error} If the request fails and there is no cache available.
+   */
+  protected async refreshTrustBoundary(
+    tokens: Credentials,
+  ): Promise<TrustBoundaryData | null> {
+    if (!this.trustBoundaryEnabled) {
+      return null;
+    }
+
+    if (this.universeDomain !== DEFAULT_UNIVERSE) {
+      // Skipping check for non-default universe domain as this feature is only supported in GDU
+      return null;
+    }
+
+    const cachedTB = this.trustBoundary;
+    if (cachedTB && cachedTB.encodedLocations === NoOpEncodedLocations) {
+      return cachedTB;
+    }
+
+    const trustBoundaryUrl = await this.getTrustBoundaryUrl();
+    if (!trustBoundaryUrl) {
+      return null;
+    }
+
+    const accessToken = tokens.access_token;
+
+    if (!accessToken || this.isExpired(tokens)) {
+      throw new Error(
+        'TrustBoundary: Error calling lookup endpoint without valid access token',
+      );
+    }
+
+    const headers = new Headers({
```

Comment:
Can we use addSharedMetadataHeaders here?
This way the subsequent calls to the allowed location endpoint will have trust boundary header too.